src/tools/actors/ActorExecutionTrace.java

```diff
@@ -1,84 +1,307 @@
 package tools.actors;
 
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.OutputStreamWriter;
+import java.lang.management.ManagementFactory;
+import java.lang.management.MemoryMXBean;
+import java.nio.ByteBuffer;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map.Entry;
+import java.util.concurrent.ArrayBlockingQueue;
+
+import javax.management.Notification;
+import javax.management.NotificationEmitter;
+import javax.management.NotificationListener;
+import javax.management.openmbean.CompositeData;
+
+import com.oracle.truffle.api.CompilerDirectives.TruffleBoundary;
+import com.sun.management.GarbageCollectionNotificationInfo;
+
+import som.VM;
 import som.VmSettings;
 import som.interpreter.actors.Actor;
+import som.interpreter.actors.Actor.ActorProcessingThread;
+import som.interpreter.actors.Actor.Mailbox;
 import som.interpreter.actors.EventualMessage;
 import som.interpreter.actors.SFarReference;
 import som.vm.ObjectSystem;
-import tools.ObjectBuffer;
+import som.vmobjects.SClass;
+import tools.debugger.FrontendConnector;
 
 public class ActorExecutionTrace {
 
   private static final int MSG_BUFFER_SIZE = 128;
+  private static final int BUFFER_POOL_SIZE = Runtime.getRuntime().availableProcessors() * 4;
 
-  /** Access to this data structure needs to be synchronized. */
-  private static final ObjectBuffer<ObjectBuffer<SFarReference>> createdActorsPerThread =
-      VmSettings.ACTOR_TRACING ? new ObjectBuffer<>(VmSettings.NUM_THREADS) : null;
+  private static final MemoryMXBean mbean = ManagementFactory.getMemoryMXBean();
+  private static final List<java.lang.management.GarbageCollectorMXBean> gcbeans = ManagementFactory.getGarbageCollectorMXBeans();
 
-  /** Access to this data structure needs to be synchronized. Typically via {@link createdActorsPerThread} */
-  private static final ObjectBuffer<ObjectBuffer<ObjectBuffer<EventualMessage>>> messagesProcessedPerThread =
-      VmSettings.ACTOR_TRACING ? new ObjectBuffer<>(VmSettings.NUM_THREADS) : null;
+  public static final ArrayBlockingQueue<ByteBuffer> bufferPool = new ArrayBlockingQueue<ByteBuffer>(BUFFER_POOL_SIZE);
+  public static final ArrayBlockingQueue<ByteBuffer> writerQueue = new ArrayBlockingQueue<ByteBuffer>(BUFFER_POOL_SIZE);
+  public static final HashMap<Short, String> symbolQueue = new HashMap<>();
```

Comment:
hm, why a hashmap, why not an array? is this sparse? what are theres symbols? the message symbols?